### PR TITLE
Fixes #265: handle conversion to Neo4j map

### LINKED ISSUE #265: Unwind on array of objects crashes
Hi.
Here is my code I want to run and do writing in Neo4j and I get an error.
I don't know from where it takes "type" field name. I guess the problem is coming from UNWIND on array of objects.
I guess it does not work as expected.

```
sparkVersion = "2.4.7"
scalaVersion = "2.12.12"
neo4jConnectorVersion = "4.0.0"
jdkVersion = "14"
```
```
case class User(id: String, firstName: String, lastName: String, gender: String, birthday: String, email: String, arrayOfObjects: Array[Obj])
case class Obj(objStringField1: String, objStringField2: String)
```
Smth like this:
```
{
  "id": "df25368a-49a1-4843-9d49-dd7de700763e",
  "firstName": "Rolfe",
  "lastName": "Noyce",
  "gender": "MALE",
  "birthday": "2000-12-19T04:00:00.000+04:00",
  "email": "rnoyce1@google.co.uk",
  "arrayOfObjects": [
    {
      "objStringField1": "test1",
      "objStringField2": "test2"
    }
  ]
}
```

```
    users.write
      .format("org.neo4j.spark.DataSource")
      .option("database", "{my-db}")
      .option("url", "bolt://{my-neo4j-server}")
      .option("authentication.basic.username", "{my-username}")
      .option("authentication.basic.password", "{my-password}")
      .mode(SaveMode.Overwrite)
      .option("query",
        """
          |UNWIND $event.arrayOfObjects as obj
          |MERGE (u:User {id: event.id, firstName: event.firstName, lastName: event.lastName, gender: event.gender, birthday: event.birthday, email: event.email, userProp: obj.objStringField1})
          |""".stripMargin)
      .save()
```
It crashes. Here is the stacktrace:
```
Caused by: java.lang.IllegalArgumentException: Field "type" does not exist.
Available fields: objStringField1, objStringField2
    at org.apache.spark.sql.types.StructType.$anonfun$fieldIndex$1(StructType.scala:305)
    at scala.collection.immutable.Map$Map2.getOrElse(Map.scala:222)
    at org.apache.spark.sql.types.StructType.fieldIndex(StructType.scala:303)
    at org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema.fieldIndex(rows.scala:187)
    at org.neo4j.spark.util.Neo4jUtil$.convertFromSpark(Neo4jUtil.scala:171)
    at org.neo4j.spark.util.Neo4jUtil$.$anonfun$convertFromSpark$1(Neo4jUtil.scala:197)
    at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:285)
    at scala.collection.Iterator.foreach(Iterator.scala:943)
    at scala.collection.Iterator.foreach$(Iterator.scala:943)
    at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
    at scala.collection.IterableLike.foreach(IterableLike.scala:74)
    at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
    at org.apache.spark.sql.catalyst.util.ArrayDataIndexedSeq.foreach(ArrayData.scala:205)
    at scala.collection.TraversableLike.map(TraversableLike.scala:285)
    at scala.collection.TraversableLike.map$(TraversableLike.scala:278)
    at org.apache.spark.sql.catalyst.util.ArrayDataIndexedSeq.map(ArrayData.scala:205)
    at org.neo4j.spark.util.Neo4jUtil$.convertFromSpark(Neo4jUtil.scala:197)
    at org.neo4j.spark.service.Neo4jWriteMappingStrategy.$anonfun$query$1(MappingService.scala:125)
    at org.neo4j.spark.service.Neo4jWriteMappingStrategy.$anonfun$query$1$adapted(MappingService.scala:123)
    at scala.collection.TraversableLike.$anonfun$flatMap$1(TraversableLike.scala:292)
    at scala.collection.immutable.Range.foreach(Range.scala:158)
    at scala.collection.TraversableLike.flatMap(TraversableLike.scala:292)
    at scala.collection.TraversableLike.flatMap$(TraversableLike.scala:289)
    at scala.collection.AbstractTraversable.flatMap(Traversable.scala:108)
    at org.neo4j.spark.service.Neo4jWriteMappingStrategy.query(MappingService.scala:123)
    at org.neo4j.spark.service.Neo4jWriteMappingStrategy.query(MappingService.scala:20)
    at org.neo4j.spark.service.MappingService.convert(MappingService.scala:234)
    at org.neo4j.spark.writer.Neo4jDataWriter.write(Neo4jDataWriter.scala:41)
    at org.neo4j.spark.writer.Neo4jDataWriter.write(Neo4jDataWriter.scala:20)
    at org.apache.spark.sql.execution.datasources.v2.DataWritingSparkTask$.$anonfun$run$2(WriteToDataSourceV2Exec.scala:118)
    at org.apache.spark.util.Utils$.tryWithSafeFinallyAndFailureCallbacks(Utils.scala:1394)
    at org.apache.spark.sql.execution.datasources.v2.DataWritingSparkTask$.run(WriteToDataSourceV2Exec.scala:116)
    at org.apache.spark.sql.execution.datasources.v2.WriteToDataSourceV2Exec.$anonfun$doExecute$2(WriteToDataSourceV2Exec.scala:67)
    at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
    at org.apache.spark.scheduler.Task.run(Task.scala:123)
    at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:411)
    at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
    at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:414)
```

### PR DESCRIPTION
**Pull request by utnaf**
Fixes #265

* added conversion to Neo4j simple map
* upgraded to testcontainers 1.15.0
* forced "jackson-annotations" dependancy version to fix issue with testcontainers

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -355,6 +355,11 @@
 <!--            <version>1.7.30</version>-->
 <!--            <scope>test</scope>-->
 <!--        </dependency>-->
+        <dependency>
```
Comment (conker84) [inline pom.xml:358]: is this necessary?

File: src/test/scala/org/neo4j/spark/DataSourceWriterTSE.scala [hunk 2]
```diff
@@ -1249,4 +1249,40 @@ class DataSourceWriterTSE extends SparkConnectorScalaBaseTSE {
...
+                    |  "arrayOfObjects": [
+                    |    {
+                    |      "objStringField1": "test1",
+                    |      "objStringField2": "test2"
+                    |    }
+                    |  ]
+                    |}""".stripMargin
+    val df = ss.read.json(Seq(jsonStr).toDS)
+
+    df.show()
```
Comment (conker84) [inline src/test/scala/org/neo4j/spark/DataSourceWriterTSE.scala]: can you please remove this?

**Review by conker84 — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (utnaf) [reply pom.xml:358]: yes, see comment on the other pr

File: src/test/scala/org/neo4j/spark/DataSourceWriterTSE.scala (hunk 2 above)
Comment (utnaf) [reply src/test/scala/org/neo4j/spark/DataSourceWriterTSE.scala]: yup

File: pom.xml (hunk 1 above)
Comment (conker84) [reply pom.xml:358]: please add it as `test` scope and add a comment on it in order to define why we force this resolution

Comment (utnaf) [reply pom.xml:358]: this is actually a dependancy that is already there, I'm just forcing the latest version

**Review by conker84 — APPROVED** [review]

**Comment by conker84** [comment]
@utnaf thank you!

**Pull request merged**